src/linter_test.cc

```diff
@@ -16,17 +16,39 @@
 
 #include "src/linter.h"
 
+#include <iostream>
 #include <string>
 #include <fstream>
 #include <sstream>
 #include <cstdio>
 
 #include "gtest/gtest.h"
+#include "absl/strings/string_view.h"
+#include "src/lint_errors.h"
+#include "zetasql/base/status.h"
+#include "zetasql/base/status_macros.h"
 
 namespace zetasql::linter {
 
 namespace {
 
+// TODO(orhanuysal): add advanced checks.
+// currently, these tests just check if the result is ok
+// number of lint errors or lint error itself is not checked
+
+void Test_if(bool expect,
```

Comment:
If you make checks to return the LinterResult, this helper test function will be simpler: it can accept than StatusOr<LinterResult>

ASSERT_OK_AND_ASSIGN(LinterResult check, CheckLineLenght(...));
but it is for some reason not open sourced as part of gmock library (however, i see copies of it in other open sourced projects from Google). You could do something similar.

Reply:
I changed back to old version. with .ok() function there is not much difference with current master right now (except formatting, this version is now Google style clang-format)